processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -919,24 +914,21 @@ public void loadDimensionIterable(
       Map<String, ColumnCapabilitiesImpl> oldColumnCapabilities
   )
   {
-    DimensionData prevDimensionData = dimensions.get();
-    DimensionData dimensionData = prevDimensionData.clone();
-    if (dimensionData.size() != 0) {
-      throw new ISE("Cannot load dimension order when existing order[%s] is not empty.", dimensionData.getDimensionDescs().keySet());
-    }
-    for (String dim : oldDimensionOrder) {
-      if (dimensionData.getDimensionDesc(dim) == null) {
-        ColumnCapabilitiesImpl capabilities = oldColumnCapabilities.get(dim);
-        dimensionData.putCapabilities(dim, capabilities);
-        DimensionHandler handler = DimensionHandlerUtils.getHandlerFromCapabilities(dim, capabilities, null);
-        dimensionData.addNewDimension(dim, capabilities, handler);
-      }
-    }
-
-    while (!dimensions.compareAndSet(prevDimensionData, dimensionData)) {
+    DimensionData prevDimensionData, dimensionData;
+    do {
       prevDimensionData = dimensions.get();
-      dimensionData.rebase(prevDimensionData);
-    }
+      dimensionData = prevDimensionData.clone();
+      if (dimensionData.size() != 0) {
+        throw new ISE("Cannot load dimension order when existing order[%s] is not empty.", dimensionData.getDimensionDescs().keySet());
```

Comment:
Line longer than 120 columns. If you use IDE for developing Druid, please set up the ruler at 120 columns to avoid adding such code.